File: CS/BackgroundGeolocation.Mobile/Geolocation/GeolocationJsonObject.cs
using Newtonsoft.Json;

namespace BackgroundGeolocation.Module.Mobile.Services {
    public class Coords {
        [JsonProperty("latitude")]
        public double Latitude { get; set; }

        [JsonProperty("longitude")]
        public double Longitude { get; set; }

        [JsonProperty("accuracy")]
        public double Accuracy { get; set; }

        [JsonProperty("speed")]
        public double Speed { get; set; }

        [JsonProperty("heading")]
        public double Heading { get; set; }

        [JsonProperty("altitude")]
        public double Altitude { get; set; }
    }

    public class Activity {

        [JsonProperty("type")]
        public string Type { get; set; }

        [JsonProperty("confidence")]
        public int Confidence { get; set; }
    }

    public class Battery {

        [JsonProperty("is_charging")]
        public bool IsCharging { get; set; }

        [JsonProperty("level")]
        public double Level { get; set; }
    }

    public class Extras {
    }

    public class Location {

        [JsonProperty("event")]
        public string Event { get; set; }

        [JsonProperty("is_moving")]
        public bool IsMoving { get; set; }

        [JsonProperty("uuid")]
        public string Uuid { get; set; }

        [JsonProperty("timestamp")]
        public string Timestamp { get; set; }

        [JsonProperty("odometer")]
        public double Odometer { get; set; }

        [JsonProperty("coords")]
        public Coords Coords { get; set; }

        [JsonProperty("activity")]
        public Activity Activity { get; set; }

        [JsonProperty("battery")]
        public Battery Battery { get; set; }

        [JsonProperty("extras")]
        public Extras Extras { get; set; }
    }

    public class Device {

        [JsonProperty("model")]
        public string Model { get; set; }

        [JsonProperty("platform")]
        public string Platform { get; set; }

        [JsonProperty("uuid")]
        public string Uuid { get; set; }

        [JsonProperty("version")]
        public string Version { get; set; }

        [JsonProperty("manufacturer")]
        public string Manufacturer { get; set; }

        [JsonProperty("framework")]
        public string Framework { get; set; }
    }

    public class GeolocationJsonObject {

        [JsonProperty("location")]
        public Location Location { get; set; }

        [JsonProperty("device")]
        public Device Device { get; set; }
    }
}

File: VB/BackgroundGeolocation.Mobile/Geolocation/GeolocationJsonObject.vb
Imports Newtonsoft.Json

Namespace BackgroundGeolocation.Module.Mobile.Services
	Public Class Coords
		<JsonProperty("latitude")>
		Public Property Latitude() As Double

		<JsonProperty("longitude")>
		Public Property Longitude() As Double

		<JsonProperty("accuracy")>
		Public Property Accuracy() As Double

		<JsonProperty("speed")>
		Public Property Speed() As Double

		<JsonProperty("heading")>
		Public Property Heading() As Double

		<JsonProperty("altitude")>
		Public Property Altitude() As Double
	End Class

	Public Class Activity

		<JsonProperty("type")>
		Public Property Type() As String

		<JsonProperty("confidence")>
		Public Property Confidence() As Integer
	End Class

	Public Class Battery

		<JsonProperty("is_charging")>
		Public Property IsCharging() As Boolean

		<JsonProperty("level")>
		Public Property Level() As Double
	End Class

	Public Class Extras
	End Class

	Public Class Location

		<JsonProperty("event")>
		Public Property [Event]() As String

		<JsonProperty("is_moving")>
		Public Property IsMoving() As Boolean

		<JsonProperty("uuid")>
		Public Property Uuid() As String

		<JsonProperty("timestamp")>
		Public Property Timestamp() As String

		<JsonProperty("odometer")>
		Public Property Odometer() As Double

		<JsonProperty("coords")>
		Public Property Coords() As Coords

		<JsonProperty("activity")>
		Public Property Activity() As Activity

		<JsonProperty("battery")>
		Public Property Battery() As Battery

		<JsonProperty("extras")>
		Public Property Extras() As Extras
	End Class

	Public Class Device

		<JsonProperty("model")>
		Public Property Model() As String

		<JsonProperty("platform")>
		Public Property Platform() As String

		<JsonProperty("uuid")>
		Public Property Uuid() As String

		<JsonProperty("version")>
		Public Property Version() As String

		<JsonProperty("manufacturer")>
		Public Property Manufacturer() As String

		<JsonProperty("framework")>
		Public Property Framework() As String
	End Class

	Public Class GeolocationJsonObject

		<JsonProperty("location")>
		Public Property Location() As Location

		<JsonProperty("device")>
		Public Property Device() As Device
	End Class
End Namespace
